From 4ba49015ca1ad98c03a209a11862f8e00d00fbd0 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Wed, 24 Aug 2016 16:19:36 +0200
Subject: [PATCH 02/23] Use strdup() if offset are used

Strings with an offset to the original starting point must be copied
because otherwise they cannot be properly freed later.
---
 library/adenroll.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/library/adenroll.c b/library/adenroll.c
index d1020e9..05885d0 100644
--- a/library/adenroll.c
+++ b/library/adenroll.c
@@ -1318,9 +1318,9 @@ load_keytab_entry (krb5_context k5,
 
 		} else if (!enroll->host_fqdn && _adcli_str_has_prefix (name, "host/") && strchr (name, '.')) {
 			/* Skip host/ prefix */
-			enroll->host_fqdn = name + 5;
-			_adcli_info ("Found host qualified name in keytab: %s", name);
-			name = NULL;
+			enroll->host_fqdn = strdup (name + 5);
+			return_val_if_fail (enroll->host_fqdn != NULL, FALSE);
+			_adcli_info ("Found host qualified name in keytab: %s", enroll->host_fqdn);
 		}
 	}
 
-- 
2.14.4

